lib/Dialect/OM/Transforms/FreezePaths.cpp

```diff
@@ -274,6 +298,9 @@ LogicalResult PathVisitor::run(ModuleOp module) {
       } else if (auto path = dyn_cast<EmptyPathOp>(op)) {
         if (failed(process(path)))
           return WalkResult::interrupt();
+      } else if (auto listCreate = dyn_cast<ListCreateOp>(op)) {
```

Comment:
Was originally wondering if this could be calling `updateType` on the operands/results or something, but then....

Reply:
> Was originally wondering if this could be calling updateType on the operands/results or something, but then....

The operands to the list_create should be updated by the pass already, it's the list_create return type that needs to be updated, but you can't update an operation's return type in place.

> More immediately, what about List<List<Path>>?

Let me check this case, we probably do need to look recursively through aggregates.

> I was wondering about other types, like Tuple

I don't think we need to worry about the other aggregates, I think we actually should be able to remove them in the future and just focus on List.

Reply:
Sounds good, thanks!

Reply:
Please take a look after the last push. I found some neat helpers which makes the logic work with potentially nested lists in a pretty nice way. I think it's actually much nicer this way anyway :)